warehouse/query-core/src/main/java/datawave/query/function/DocumentProjection.java

```diff
@@ -38,40 +38,43 @@ public class DocumentProjection implements DocumentPermutation {
     private boolean trackSizes = true;
 
     public DocumentProjection() {
-        this(false, false);
+        this.includeGroupingContext = false;
+        this.reducedResponse = false;
+        this.projection = new Projection();
     }
 
     public DocumentProjection(boolean includeGroupingContext, boolean reducedResponse) {
-        this(includeGroupingContext, reducedResponse, true);
+        this.includeGroupingContext = includeGroupingContext;
+        this.reducedResponse = reducedResponse;
+        this.projection = new Projection();
     }
 
-    public DocumentProjection(boolean includeGroupingContext, boolean reducedResponse, boolean trackSizes) {
+    public DocumentProjection(boolean includeGroupingContext, boolean reducedResponse, boolean trackSizes, Set<String> projections,
```

Comment:
This is the actual constructor we want to use long term, deprecate the others.

Reply:
rgr